arrow-schema/src/schema.rs

```diff
@@ -140,6 +140,15 @@ impl From<Fields> for SchemaBuilder {
     }
 }
 
+impl From<&Schema> for SchemaBuilder {
+    fn from(value: &Schema) -> Self {
+        Self {
+            fields: value.fields.to_vec(),
+            metadata: value.metadata.clone(),
+        }
```

Comment:
This could be slightly more concise 🤔 

```suggestion
        Self::from(value.clone())
```